### DISCUSSION TITLE
How does one go to the next unused group?

### DISCUSSION THREAD

**Discussion by developer1105** [qtile/qtile#3667] [Q&A]
Let's suppose I have a terminal open in tag 1. I would like to assign a keybinding to go to the first unused tag, in this case, the tag 2. If I had tags 1 to 5 filled with apps, I would like the same keybinding to jump to tag 6.

**Comment by elParaguayo** [comment]
I don't think we have a built-in command for this so something like this might work:
``` python
@lazy.function
def go_to_empty_group(qtile):
    for group in qtile.groups:
        if not group.windows:
            qtile.current_screen.set_group(group)
            break

# Then just bind go_to_empty_group() to a key
```

**Reply by developer1105** [reply]
It worked fine! 

The trick is that in the keybinding one should not use "lazy.go_to_empty_group()" but rather only go_to_empty_group() for it to work. 

It kind of diverges from all the other keybindings (which usually start with "lazy.something") but it does exactly what I expected, thanks!

**Reply by elParaguayo** [reply]
You don't need the `lazy` prefix as you've decorated the function with `@lazy.function`.

Alternatively, you can remove the decorator and bind `lazy.function(go_to_empty_group)`.

**Comment by asoymos** [comment]
Hi, I was searching and was about to ask when I found this discussion. Would recommend to change title to group instead of tag so others can find this in google and avoid same questions. Thank you.

**Comment by asoymos** [comment]
I have set different groups per monitor as specified [here](https://docs.qtile.org/en/stable/manual/faq.html#how-can-i-get-my-groups-to-stick-to-screens), and mixing the [answer](https://github.com/qtile/qtile/discussions/3667#discussioncomment-3048403) in here with the logic proposed in my own [question](https://github.com/qtile/qtile/discussions/4722) I came up to this version for going to the first unused group in current screen:
```python
@lazy.function
def go_to_first_empty_group_in_same_screen(qtile):
    # Get the current screen index
    screen = qtile.current_screen.index
    # Find all groups marked for that screen or that have no affinity and that not have windows
    screen_groups = [g for g in qtile.groups if g.screen_affinity in [screen, None] and not g.windows]
    # Go to first of that groups 
    for group in screen_groups:
        qtile.current_screen.set_group(group)
        break
```
